### PR TITLE
cloudwatch api backoff

### PR DESCRIPTION
**Pull request by paul-krohn**
The CloudWatch API has a limit, but as AWS won't tell me what it is, I can't validate that this change actually solves the problem at hand; it really is a test of whether I stop seeing the occasional error that the limit has been exceeded.

### CODE REVIEW

**Comment by jib** [comment]
+1 on the logic, but I bow to others for python-ness

**Comment by plathrop** [comment]
+1

File: leadbutt.py [hunk 1]
```diff
@@ -144,18 +162,19 @@ def leadbutt(config_file, cli_options, verbose=False, **kwargs):
...
+                metric_name=metric_name,
+                namespace=metric['Namespace'],
+                statistics=metric['Statistics'],
                 dimensions=metric['Dimensions'],
-                unit=unit,
+                unit=unit
             )
-            # sys.stderr.write('{} {}\n'.format(options['Count'], len(results)))
             output_results(results, this_metric, options)
+            time.sleep(float(kwargs.get('interval', 0)) / 1000.0)
```
Comment (hanpeter) [inline leadbutt.py:178]: The overall logic is complicating enough that one has to read the code 2-3 times. The dual usage of `-i`, while very clever, will catch someone off guard. Let's comment this.

**Comment by hanpeter** [comment]
LGTM. Requesting some more comments, but that can wait until the final version.

File: leadbutt.py (hunk 1 above)
Comment (paul-krohn) [inline leadbutt.py:178]: There is a comment in the docstring (https://github.com/krux/cloudwatch-to-graphite/pull/7/files#diff-75d44ccbc6fb95936c59f2fb20cbb35cR9), which ... doubles as the options definition.

**Pull request merged**